lexibank_kpaamcam.py

```diff
@@ -41,33 +47,29 @@ def cmd_makecldf(self, args):
         """
         Convert the raw data to a CLDF dataset.
         """
-
-        # Write source
         args.writer.add_sources()
 
-        # Write languages
-        languages = args.writer.add_languages(lookup_factory='Name')
-
-        # Write concepts
         concepts = {}
         for concept in self.concepts:
             idx = concept['NUMBER']+'_'+slug(concept['ENGLISH'])
+            similarity = int(concept['SIMILARITY'] or 4)
             args.writer.add_concept(
-                    ID=idx,
-                    Name=concept['ENGLISH'],
-                    PartOfSpeech=concept['POS'],
-                    )
+                ID=idx,
+                Name=concept['ENGLISH'],
+                Concepticon_ID=concept['CONCEPTICON_ID'] if similarity <= 2 else None,
+                Concepticon_Gloss=concept['CONCEPTICON_GLOSS'] if similarity <= 2 else None,
+            )
             concepts[concept['ENGLISH']] = idx
 
-        # Write forms
-        wl = Wordlist(self.raw_dir.joinpath('oneEntryPerRow-wordlist.tsv').as_posix())
-        for idx in progressbar(wl):
-            #print(languages[wl[idx, 'doculect']])
-            args.writer.add_forms_from_value(
-                    Value=wl[idx, 'value'],
-                    Language_ID=languages[wl[idx, 'doculect']],
-                    Parameter_ID=concepts[wl[idx, 'concept']],
-                    Reflex_ID=wl[idx, 'reflex_id'],
-                    Source=[]
-                    )
-
+        for i, entry in enumerate(self.raw_dir.read_csv('parallel_lists.csv', dicts=True)):
+            for lid, glottocode in speakerInfos.items():
+                if i == 0:
+                    args.writer.add_language(ID=lid, Name=lid, Glottocode=glottocode)
+                if not entry[lid].strip():
+                    continue
+                r = args.writer.add_forms_from_value(
+                    Value=entry[lid],
+                    Language_ID=lid,
+                    Parameter_ID=concepts[entry['Concept']],
+                    Source=['Tschonghongei:2021']
+                )
```

Comment:
Here's how to add the lexical data to the CLDF dataset without intermediate `lingpy.Wordlist`. It's actually simpler :)